consensus/replay_test.go

```diff
@@ -83,7 +83,11 @@ func startNewStateAndWaitForBlock(t *testing.T, consensusReplayConfig *cfg.Confi
 
 	err := cs.Start()
 	require.NoError(t, err)
-	defer cs.Stop()
+	defer func() {
+		if err := cs.Stop(); err != nil {
+			t.Error(err)
+		}
+	}()
```

Comment:
Should this be `t.Cleanup()`?

Reply:
Surprisingly I get a bunch of test failures when I use `t.Cleanup` here. Still looking to find the reason

Reply:
Oh. Because `t.Cleanup()` only runs when the test ends, while `defer` runs when `startNewStateAndWaitForBlock` returns. This is fine.